src/frontend/hooks/useNavigationWithTypes.ts

```diff
@@ -0,0 +1,4 @@
+import { useNavigation as useNavigationNative } from "@react-navigation/native";
+import { StackNavProp } from "../Navigation/AppStack";
+
+export const useNavigation = () => useNavigationNative<StackNavProp>();
```

Comment:
the `useNavigation` hook utilizes native navigation AND is passed all the route name and params for typesafety. useNavigation should be imported from here, not `@react-navigation/native`